test commit

OurnoFrontendDev opened the pull request.

OurnoFrontendDev committed:
test commit

OurnoFrontendDev committed:
Corrected edits

OurnoFrontendDev committed:
Corrected edits 1.0

OurnoFrontendDev committed:
Corrected edits 1.1

danimaxi54 commented on .github/workflows/ci-test.yml:
```diff
@@ -0,0 +1,22 @@
+  push:
+    branches: [ Comment ]
+  pull_request:
+    branches: [ Comment ]
```
в любой ветке пайплайн должен срабатывать, а ты зашил на comment ветку

danimaxi54 commented on src/App.tsx:
```diff
@@ -0,0 +1,26 @@
+      <AppContainer>
+        <CommentWrapper>
+          <Header>Comments</Header>
+          <CommentForm avatar={'/img/avatar1.jpg'} username={'John Doe'} />
```
у тебя иконки должны импортироваться как react компоненты

danimaxi54 commented on src/GlobalStyle.ts:
```diff
@@ -0,0 +1,19 @@
+import { createGlobalStyle } from 'styled-components';
+
+export const GlobalStyle = createGlobalStyle`
+    
```
лишняя пустая линия

danimaxi54 commented on src/app/providers/StoreProvider/StoreProvider.tsx:
```diff
@@ -0,0 +1,11 @@
+  children: ReactNode;
+}
+
+export const StoreProvider: React.FC<StoreProviderProps> = ({ children }) => {
```
тело - фигурные скобки тут не нужны, можно без них

danimaxi54 commented on src/components/CommentItem.tsx:
```diff
@@ -0,0 +1,123 @@
+  IconButton,
+  IconContainer,
+  UserName,
+} from './styledComponents/StyledComponents';
```
а зачем папка styled components?
у тебя в папке components должны лежать компоненты

danimaxi54 commented on src/components/CommentItem.tsx:
```diff
@@ -0,0 +1,123 @@
+} from './styledComponents/StyledComponents';
+import { timeAgo } from '../utils/timeAgo';
+
+const sizeIcon = 20;
```
зачем?

danimaxi54 commented on src/components/CommentItem.tsx:
```diff
@@ -0,0 +1,123 @@
+import { timeAgo } from '../utils/timeAgo';
+
+const sizeIcon = 20;
+const iconColor = '#00bfff';
```
const REPLY_ICON_COLOR 
как минимум
а зачем здесь переменная вообще?

danimaxi54 commented on src/components/CommentItem.tsx:
```diff
@@ -0,0 +1,123 @@
+const iconColor = '#00bfff';
+
+export const CommentItem: React.FC<{
+  dislike: number;
```
пропсы нужно выносить в отдельный тип 
type Props или interface Props

danimaxi54 commented on src/components/CommentItem.tsx:
```diff
@@ -0,0 +1,123 @@
+    }
+  }, [comment.id, like]);
+
+  const handleDisLike = useCallback(async () => {
```
зачем здесь useCallback? какая ценность в нем?

danimaxi54 commented on src/components/CommentItem.tsx:
```diff
@@ -0,0 +1,123 @@
+  const [showReplyForm, setShowReplyForm] = useState(false);
+
+  const handleLike = useCallback(async () => {
+    const Like = like === 1 ? 0 : 1;
```
почему переменная названа с большой буквы?

danimaxi54 commented on src/components/CommentItem.tsx:
```diff
@@ -0,0 +1,123 @@
+    }
+  }, [comment.id]);
+
+  const avatarSrc = useMemo(
```
useMemo зачем здесь? 
иконки должны импортироваться как React компоненты

danimaxi54 commented on src/components/CommentItem.tsx:
```diff
@@ -0,0 +1,123 @@
+
+  const handleReplyFormClose = useCallback(() => {
+    setShowReplyForm(false);
+  }, []);
```
пустая линия должны быть между функциями и jsx

danimaxi54 commented on src/components/CommentItem.tsx:
```diff
@@ -0,0 +1,123 @@
+  return (
+    <CommentContentContainer>
+      <CommentItemAvNameContainer>
+        <Avatar src={avatarSrc} alt={'Bra'}></Avatar>
```
children нет, поэтому </Avatar> не нужен в конце

danimaxi54 commented on src/components/CommentItem.tsx:
```diff
@@ -0,0 +1,123 @@
+      </CommentItemAvNameContainer>
+      <CommentForm
+        $parentId={comment.parentId}
+        dangerouslySetInnerHTML={{ __html: comment.content }}
```
запрет на такое 
в этом кейсе точно нет

danimaxi54 commented on src/components/CommentItem.tsx:
```diff
@@ -0,0 +1,123 @@
+        $parentId={comment.parentId}
+        dangerouslySetInnerHTML={{ __html: comment.content }}
+      />
+      <IconContainer withFrameButtonGroup={true}>
```
просто withFrameButtonGroup
переменная не отражает названия, уже какой по счету проект пишешь? и я про тоже пишу

danimaxi54 commented on src/components/CommentItem.tsx:
```diff
@@ -0,0 +1,123 @@
+          <FcDislike size={sizeIcon} />
+          {dislike}
+        </IconButton>
+        <IconButton onClick={() => setShowReplyForm(!showReplyForm)}>
```
setShowReplyForm(prev => !prev)

danimaxi54 commented on src/components/CommentItem.tsx:
```diff
@@ -0,0 +1,123 @@
+  IconContainer,
+  UserName,
+} from './styledComponents/StyledComponents';
+import { timeAgo } from '../utils/timeAgo';
```
создай папку utils/format/date/index.tsx

danimaxi54 commented on src/components/CommentItem.tsx:
```diff
@@ -0,0 +1,123 @@
+
+      {replies.map((reply) => (
+        <CommentItem
+          key={reply.id}
```
зачем каждый проп передавать?
если можно развернуть {...reply}

danimaxi54 commented on src/components/CommentsForm.tsx:
```diff
@@ -0,0 +1,110 @@
+
+  const [content, setContent] = useState(initialContent || '');
+
+  const isDisabledFormAdd = content.length === 0;
```
isDisabledAddCommentBtn

danimaxi54 commented on src/components/CommentsForm.tsx:
```diff
@@ -0,0 +1,110 @@
+
+  const isDisabledFormAdd = content.length === 0;
+
+  const handleTextFormatClick = useCallback(
```
useCallback зачем

danimaxi54 commented on src/components/CommentsForm.tsx:
```diff
@@ -0,0 +1,110 @@
+      <CommentFormAvNameContainer>
+        <Avatar src={avatar} alt={username} />
+        <Username>{'John Doe'}</Username>
+      </CommentFormAvNameContainer>
```
CommentFormAvNameContainer

непонятное название компонента

danimaxi54 commented on src/components/CommentsForm.tsx:
```diff
@@ -0,0 +1,110 @@
+      </CommentFormAvNameContainer>
+      <form onSubmit={handleSubmit}>
+        <TextArea
+          id="commentTextarea"
```
id зачем?

danimaxi54 commented on src/components/CommentsForm.tsx:
```diff
@@ -0,0 +1,110 @@
+          <IconContainer withFrameButtonGroup={false}>
+            <IconButton
+              title="Bold"
+              onClick={() => handleTextFormatClick('<b>')}
```
handleTextFormatClick('bold')

danimaxi54 commented on src/components/CommentsForm.tsx:
```diff
@@ -0,0 +1,110 @@
+            </IconButton>
+            <IconButton
+              title="Italic"
+              onClick={() => handleTextFormatClick('<em>')}
```
handleTextFormatClick('cursive')

danimaxi54 commented on src/components/CommentsList.tsx:
```diff
@@ -0,0 +1,108 @@
+  const { data: comments } = useGetCommentsQuery('');
+  const [deleteComment] = useDeleteCommentMutation();
+
+  const itemsHeights = useRef<Record<number, number>>({});
```
commentsHeights

danimaxi54 commented on src/components/CommentsList.tsx:
```diff
@@ -0,0 +1,108 @@
+    [deleteComment],
+  );
+
+  const getReplies = useCallback(
```
useCallback зачем

danimaxi54 commented on src/components/CommentsList.tsx:
```diff
@@ -0,0 +1,108 @@
+
+  const getReplies = useCallback(
+    (parentId: string | null) => {
+      return comments?.filter((comment) => comment.parentId === parentId) || [];
```
(comments || []).filter((comment) => comment.parentId === parentId)

danimaxi54 commented on src/components/CommentsList.tsx:
```diff
@@ -0,0 +1,108 @@
+    [comments],
+  );
+
+  const Row: React.FC<{ index: number; style: object }> = ({
```
что за название компонента такое? 
CommentItem 
далее, компонент должен жить в отдельной папке 
создай отдельный тип под Props 
style: object с чего вдруг? styles: CSSProperties, хотя бы так

danimaxi54 commented on src/components/CommentsList.tsx:
```diff
@@ -0,0 +1,108 @@
+  }) => {
+    const comment = parentComments[index];
+    const newStyle = { ...style, width: '100%' };
+    const replies = getReplies(comment.id);
```
в useMemo

danimaxi54 commented on src/components/CommentsList.tsx:
```diff
@@ -0,0 +1,108 @@
+    }, [index, setDialogItemSize]);
+
+    return (
+      <div style={newStyle} key={comment.id}>
```
зачем тут ключ? ключи указываются на верхнем уровне
самое базовое правило react про ключи

danimaxi54 commented on src/components/CommentsList.tsx:
```diff
@@ -0,0 +1,108 @@
+    return (
+      <div style={newStyle} key={comment.id}>
+        <CommentWrapper ref={ref}>
+          <MemoizedCommentItem
```
Memoized зачем такое название в начале?

danimaxi54 commented on src/components/CommentsList.tsx:
```diff
@@ -0,0 +1,108 @@
+  };
+
+  useEffect(() => {
+    if (virtualList.current && parentComments.length > 0) {
```
virtualList.current && parentComments.length

danimaxi54 commented on src/components/ReplyForm.tsx:
```diff
@@ -0,0 +1,65 @@
+
+  const [addReplyComment] = useReplyCommentMutation();
+
+  const isDisabledReplyAdd = content.length === 0;
```
isDisabledReplyCommentBtn

danimaxi54 commented on src/components/styledComponents/StyledComponents.tsx:
```diff
@@ -0,0 +1,200 @@
+  background: none;
+  border: none;
+  cursor: pointer;
+  font-size: 1.5em;
```
зачем em

danimaxi54 commented on src/models/IComments.ts:
```diff
@@ -0,0 +1,11 @@
+export interface IComments {
```
src/types/comments.ts

danimaxi54 commented on src/utils/timeAgo.ts:
```diff
@@ -0,0 +1,19 @@
+export const timeAgo = (createdAt: string) => {
```
dayjs для этого либу юзай

OurnoFrontendDev committed:
1.Добавил срабатывание пайплайн на любой ветке;
2.Иконки отображаются с помощью компонентов, изображения берутся по Url, исправил avatar на avatarUrl;
3.Убрал лишние строки;
4.Убрал ненужные useCallback и useMemo;
5.Разнес компоненты по отдельным папкам, добавил их стили в отдельный файл в папке компоненты;
6.Переименовал названия компонентов и переменных на более понятные;
7.CommentItem.tsx - использовал rest при передаче props
8.Добавил в utils format/date/index.ts => использовал dayjs
9.Добавил отступ комментариев от скролла
10.К кнопкам B,I и Comment добавил cursor: not-allowed при disable;
11. Добавил удаление комментария через форму
12.Пользователь не имеет больше возможности одновременно поставить лайк и дизлайк
13.Выделение тела комментария жирным или курсивом изменил с dangerouslysetinnerhtml на использование ReactMarkdown
14.Компоненту Row переименовал на RenderingCommentItem и вынес в отдельную папку
15. Изменил ширину родительского комментария
16.Создал type для props компоненты CommentItem.tsx
17.Компоненту CommentFormAvNameContainer переименовал на CommentAvatarAuthor
18. Изменил в компоненте IconButton свойство  font-size: c em  на px
19.Вынес стили Header в отдельную папку
20.Вынес типы в отдельную папку types

OurnoFrontendDev committed:
1.svgLoader => iconLoader
